### ISSUE TITLE
Very poor performance when switching projects with Rancher UI

### ISSUE THREAD

**Issue by willmao** [rancher/rancher#16605]
**What kind of request is this (question/bug/enhancement/feature request):enhancement

**Steps to reproduce (least amount of steps as possible):**
1. login into rancher ui
2. select a project in the k8s cluster and go to the workload page
3. switch to another project

**Result:**
The workload page spends about 10s to fully rendered

**Other details that may be helpful:**

My project has about 80 deployments and i checked the XHR requests, the api calls to get pods and workloads are very slow, each takes 5 and 10 seconds to return the results.

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): 2.1.1
- Installation option (single install/HA): HA 

resources allocated:

resources:
  limits:
    cpu: "2"
    memory: 2Gi
  requests:
    cpu: 200m
    memory: 500Mi

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Custom
- Machine type (cloud/VM/metal) and specifications (CPU/memory): bare metal 4cores*8G
- Kubernetes version (use `kubectl version`): 1.11.3

```
Client Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.3", GitCommit:"a4529464e4629c21224b3d52edfe0ea91b072862", GitTreeState:"clean", BuildDate:"2018-09-09T18:02:47Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.3", GitCommit:"a4529464e4629c21224b3d52edfe0ea91b072862", GitTreeState:"clean", BuildDate:"2018-09-09T17:53:03Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}
```

- Docker (use `docker info`):

```
Containers: 25
 Running: 21
 Paused: 0
 Stopped: 4
Images: 59
Server Version: 17.03.2-ce
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 362
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 4ab9917febca54791c5f071a9d1f404867857fcc
runc version: 54296cf40ad8143b62dbcaa1d90e520a2136ddfe
init version: 949e6fa
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.4.0-117-generic
Operating System: Ubuntu 16.04.4 LTS
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 7.796 GiB
Name: yit-dev-infra-01
ID: DLA3:SAV5:5SQ3:LCKD:4FQ3:XG4V:JHNG:RFEY:EKRQ:2JYU:HTAA:AJZF
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 registry.docker.yit.com
 127.0.0.0/8
Registry Mirrors:
 https://nwg8wc6k.mirror.aliyuncs.com
 https://registry.docker-cn.com
Live Restore Enabled: false

WARNING: No swap limit support
```

**Comment by superseb** [comment]
Pleas supply more info, are you running a 3 node HA cluster with Rancher and a custom cluster with the 80 deployments? The Rancher HA cluster should be dedicated to running Rancher and no limits have to be set for the rancher Deployment. From there, you can start creating clusters to run your workloads.

**Comment by willmao** [comment]
thanks for your reply! we are running the Rancher HA cluster in the same k8s cluster as our workloads which created by rke. we have 15 projects and each has about 80 workloads(microservices). we did not run Rancher HA in a dedicated cluster because create a k8s cluster with rke is more flexible, compared with booting Rancher first and then create workloads cluster.

**Comment by loganhz** [comment]
Can you check if the communication between rancher server and your k8s cluster is via private network?

Can you check if the communication between kube-apiserver and k8s nodes inside your k8s cluster  is via private network?

If the communication is using public network and there might be some bandwidth limitation for the public network, it will be slow.

**Comment by willmao** [comment]
all nodes in the cluster are located in a private network, each node has a 1000 Mbps bandwidth and the latency between two nodes is less than 0.3ms.

Besides, the rancher was deployed behind a L7 load balancer

**Comment by XzenTorXz** [comment]
We have the same issue. But it seems not hardware relevant. We have 2 servers both running rancher single node setup with only themself as node. 
Both servers have the same hardware and running rancher 2.1.4 with docker 17.03.2-ce.
At installation both where running fine. But now we have one server (server 2) having this issue, where every project change takes around 30-40 seconds. Within the projects everything works fast.
Both servers have not much of load, but server 2 has constant 16% cpu load and server 1 "only" 7%

Server 2:
![grafik](https://user-images.githubusercontent.com/3819224/52288784-1b6fda00-296d-11e9-85b4-2005c8653130.png)

the rancher/coreos seem to eat the most of it, and this error log keeps repeating:

> 2019-02-05 16:41:32.020989 W | etcdserver: read-only range request "key:\"/registry/minions/node2\" " with result "range_response_count:1 size:9961" took too long (159.561641ms) to execute
2019-02-05 16:41:32.853924 W | etcdserver: read-only range request "key:\"/registry/minions/node2\" " with result "range_response_count:1 size:9961" took too long (192.194066ms) to execute
2019-02-05 16:41:35.502328 W | etcdserver: read-only range request "key:\"/registry/minions/node2\" " with result "range_response_count:1 size:9961" took too long (240.823384ms) to execute
2019-02-05 16:41:41.963160 W | etcdserver: request "header:<ID:10646624685515675968 username:\"system:node\" auth_revision:1 > txn:<compare:<target:MOD key:\"/registry/masterleases/5.9.31.70\" mod_revision:7706029 > success:<request_put:<key:\"/registry/masterleases/5.9.31.70\" value_size:65 lease:1423252648660900158 >> failure:<request_range:<key:\"/registry/masterleases/5.9.31.70\" > >>" with result "size:20" took too long (125.514073ms) to execute
2019-02-05 16:41:41.963278 W | etcdserver: read-only range request "key:\"/registry/secrets/kube-system/cronjob-controller-token-6m6hm\" " with result "range_response_count:1 size:2398" took too long (299.257983ms) to execute

Server 1:
![grafik](https://user-images.githubusercontent.com/3819224/52288745-07c47380-296d-11e9-9b93-73565b5ab2cd.png)

the etc log looks about the same, maybe half as frequent with the error

Both of the Server have also the error in the kubelet error log, every second if its relevant:

> 0205 16:47:53.307431    2523 kubelet_volumes.go:140] Orphaned pod "03e83326-28e2-11e9-9213-5404a6f17789" found, but volume paths are still present on disk : There were a total of 341 errors similar to this. Turn up verbosity to see them.

What is also worth noting: server 1 had the same issue before, that got resolved when we reinstalled everything.

**Comment by XzenTorXz** [comment]
I found the issue for our servers. Some apps producing thousands of ConfigMaps like "app.vXXXX"
I feel like those are created on crashes and when trying to recreate the pod (just a feeling). We had over 12000 such maps.
After I deleted all of the ConfigMaps the change between Projects is not as long anymore.

An example for deleting those maps:

`kubectl delete ConfigMap --namespace=longhorn-system $(kubectl get ConfigMap --namespace=longhorn-system | grep longhorn.v | cut -d' ' -f1)`


So where do these ConfigMap come from ? And how to prevent it in the future ? 
#18986 #18870 might be related

**Comment by daxmc99** [comment]
Hi @XzenTorXz,
The problem with excessive configmaps will be fixed in this PR https://github.com/rancher/rancher/pull/19079.
The issue is due to an error where rancher's internal tiller would constantly attempt to recreate a resource despite a flaw in a helm chart provided. Going forward it will be more visible in the UI when tiller fails to deploy and configmaps will be limited to 1. 
If possible, it would be helpful to know which app you observed the failures on and its configuration.

**Comment by rmweir** [comment]
@XzenTorXz We are also working to isolate slowness to only effect the projects the config maps belong too when a situation such as this arises, https://github.com/rancher/rancher/issues/18870.

**Comment by XzenTorXz** [comment]
@daxmc99 one was one of our own app and the other was longhorn (Version 0.3.0 or 0.3.1)

**Comment by junkiebev** [comment]
This was happening to me as well - the problem seemed to be the new (rancher v2.2.0) monitoring capability blowing up. Removing the feature and running this command fixed the problem:
`kubectl delete ConfigMap --namespace=cattle-prometheus $(kubectl get ConfigMap --namespace=cattle-prometheus | grep cluster-monitoring.v | cut -d' ' -f1)`

**Comment by deniseschannon** [comment]
@willmao The other 2 users in this issue have attributed their issues to excessive configmaps that aren't being cleaned up, which is something we're addressing in a future release. Can you look at your configmaps to see if that's also an issue for you?

**Comment by willmao** [comment]
@deniseschannon our cluster has 2200 configmaps, i don't know it is excessive enough to cause this issue.

**Comment by daxmc99** [comment]
@willmao From some testing I have done that might increase UI latency by 10 seconds, possibly more if they are large configmaps.

**Comment by jira-sync-svc** [comment]
➤ Jack Luo commented:

The big fix is validated in Rancher: v2.2.2-rc1 in both single installed and HA mode

see https://github.com/rancher/rancher/issues/18522#issuecomment-486481649 ( https://github.com/rancher/rancher/issues/18522#issuecomment-486481649|smart-link )  for detailed test results.

**Comment by albertmatyi** [comment]
I've just updated to `2.2.6` and the UI became responsive. We've been on `2.2.1` and sometimes the UI came to a halt, couldn't access projects, or was really slow.

Will report back shall the performance degrade again with more details about our setup
